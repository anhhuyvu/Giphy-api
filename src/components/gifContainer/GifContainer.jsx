import React from 'react';

const GifContainer = React.forwardRef(({ url, title, onclick }, ref) => {
  return (
    <div data-testid="gif-container-id" className="img-container" ref={ref} onClick={onclick}>
      <img src={url} alt={title} />
    </div>
  );
});

export default GifContainer;
