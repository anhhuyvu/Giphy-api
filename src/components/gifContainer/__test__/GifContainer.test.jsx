import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';

import GifContainer from '../GifContainer';

const props = {
  title: 'title',
  url: 'url',
  onclick: jest.fn(),
};
describe('GifContainer Component', () => {
  it('should render snapshot correctly', () => {
    const tree = renderer.create(<GifContainer {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render image correctly', () => {
    const wrapper = render(<GifContainer {...props} />);

    expect(wrapper.getByRole('img')).toBeInTheDocument();
    expect(wrapper.getByTestId('gif-container-id')).toBeInTheDocument();
  });

  it('should call function when click on image', () => {
    render(<GifContainer {...props} />);

    fireEvent.click(screen.getByRole('img'));

    expect(props.onclick).toHaveBeenCalledTimes(1);
  });
});
