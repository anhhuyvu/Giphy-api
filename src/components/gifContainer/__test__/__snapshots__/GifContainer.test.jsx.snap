// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifContainer Component should render snapshot correctly 1`] = `
<div
  className="img-container"
  data-testid="gif-container-id"
  onClick={[MockFunction]}
>
  <img
    alt="title"
    src="url"
  />
</div>
`;
